src/features/workflow/WorkflowEditor.tsx

```diff
@@ -53,6 +62,23 @@ const WorkflowEditor: React.FC<WorkflowEditorProps> = ({ runMode, run, workflow
     }
   }
 
+  // appends username, name, and meta.title to a dry run's preview, useful for
+  // display of downstream components that expect these as part of a Dataset (e.g. fullscreen body)
+  const appendRefAndMeta = (dsPreview: Dataset) => {
+    if (dsPreview) {
+      return {
+        peername: qriRef.username,
+        name: qriRef.name,
+        meta: {
+          title: 'Dry Run Data Preview'
```

Comment:
what about "Workflow Result"? What happens if the user writes a transform script that does this?

```py
def transform(ds, ctx):
  ds.set_meta("title", "some other title")
```

I'd say change the `meta` assignment to:

```tsx
return {
 // ...
  meta: dspreview.meta || { title: "Workflow Result" }
}
```